src/platform/common/configuration/service.base.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ConfigurationTarget, Uri, WorkspaceConfiguration } from 'vscode';
+import { IServiceContainer } from '../../ioc/types';
+import { IWorkspaceService } from '../application/types';
+import { JupyterSettings } from '../configSettings';
+import { isTestExecution, isUnitTestExecution } from '../constants';
+import { IConfigurationService, IWatchableJupyterSettings } from '../types';
+
+export abstract class BaseConfigurationService implements IConfigurationService {
```

Comment:
ConfigurationService was also refactored into a base class and a node/web variant.

This is because SystemVariables uses node only classes so it was split as well into a node/web variant.